### PR TITLE
feat: Validate generated sas token before publishing artifacts

### PR DESCRIPTION
**Pull request by AbelHu**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Validate generated sas token before publishing artifacts

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

Validated it in https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=68716276&view=logs&j=26d3b11b-de37-5bdb-7f8b-686d226a8fb0&t=e57474fd-ef2a-530e-8fb7-25622dad9b67 with reverting @junjiezhang1997 's fix.

**Release note**:
```
none
```

### CODE REVIEW

**Review by ShiqianTao — APPROVED** [review]

**Review by junjiezhang1997 — APPROVED** [review]

**Review by wanqingfu — APPROVED** [review]

File: vhdbuilder/packer/generate-vhd-publishing-info.sh [hunk 1]
```diff
@@ -57,6 +57,14 @@ if [ "$sas_token" == "" ]; then
 fi
 vhd_url="${STORAGE_ACCT_BLOB_URL}/${VHD_NAME}?$sas_token"
 
+echo "Testing whether the generated sas token works"
+vhd_size=$(curl -sI $vhd_url | grep -i Content-Length | awk '{print $2}')
```
Comment (alexeldeib) [inline vhdbuilder/packer/generate-vhd-publishing-info.sh:61]: should we just check the response code instead? I think even for a 401/403/404, the response size would be non-zero XML payload?

Comment (alexeldeib) [reply vhdbuilder/packer/generate-vhd-publishing-info.sh:61]:
```bash
res=$(curl -o /dev/null -w "%{http_code}" -fsSL -I "$vhd_url")
if test "$res" != "200"; then
  exit 1
fi
```

or similar?

Comment (AbelHu) [reply vhdbuilder/packer/generate-vhd-publishing-info.sh:61]: I think that it is same. This command gets the content length. I have tested that the invalid token will show empty result but the valid token will get a non-empty `integer`.

Comment (AbelHu) [reply vhdbuilder/packer/generate-vhd-publishing-info.sh:61]: ` curl -sI $vhd_url` with a valid token but a non-exist blob returns below response
```
HTTP/1.1 404 The specified blob does not exist.
Transfer-Encoding: chunked
Server: Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0
x-ms-request-id: 0da001d6-901e-000f-56a3-414af3000000
x-ms-version: 2018-11-09
x-ms-error-code: BlobNotFound
Date: Thu, 16 Feb 2023 01:09:47 GMT
```

**Pull request merged**